com.playeveryware.eos/Runtime/Core/Config.cs

```diff
@@ -324,19 +303,51 @@ protected virtual async Task ReadAsync()
         /// </summary>
         protected virtual void Read()
         {
-            // This conditional exists because writing a config file is only
-            // something that should ever happen in the editor.
-            // This is the config writing for Editor Playmode
-            // Use WriteAsync instead on Editor Not-Playmode
+            EnsureConfigFileExists(false);
+            _lastReadJsonString = FileSystemUtility.ReadAllText(FilePath);
+            JsonUtility.FromJsonOverwrite(_lastReadJsonString, this);
+        }
+
+        /// <summary>
+        /// Determines if the config file exists, and if it does not, and the
+        /// editor is running, then create the file.
+        /// </summary>
+        /// <param name="isAsync">
+        /// Indicates whether the operation should run synchronously or
+        /// asynchronously.
+        /// </param>
+        /// <returns>Task.</returns>
+        private async Task EnsureConfigFileExistsAsync(bool isAsync)
```

Comment:
I feel weird about this... could we use a Task<bool> to return if it exists or not, instead of relying on the exception? I guess this is mostly for config code, which should always exist.

Reply:
Yeah re-reading these changes they are pretty wonky. I replaced them with a better approach, which is, when an "Async" function needs to be called from a non-async context, do the following:

```csharp
public async Task DoWorkAsync()
{
    /* Some kind of async calls using the "await" keyword */
}

public void DoWork()
{
    DoWorkAsync().GetAwaiter().GetResult()
}
```

And for circumstances where the result is not void:

```csharp
public async Task<int> DoWorkAsync()
{
    /* Some kind of async calls using the "await" keyword */
}

public int DoWork()
{
    return DoWorkAsync().GetAwaiter().GetResult()
}
```

I am going to mark the other comment you made regarding this as resolved, so we can continue the conversation in a single place (this conversation)